### PR TITLE
feat(nextcloud): #DRIV-26 add new route for copying and moving files from nextcloud

### PR DESCRIPTION
**Pull request by aymericdevriesere**

### CODE REVIEW

File: src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java [hunk 1]
```diff
@@ -295,15 +302,286 @@ public Future<JsonArray> uploadFiles(UserNextcloud.TokenProvider userSession, Li
...
+                                    result.put(file, status.result());
+                                }
+                                else {
+                                    result.put(file, status.cause().getMessage());
+                                }
+                                promiseResult.complete();
+                            });
+                        }
+                        else {
+                            //TODO Gérer le cas d'import d'un dossier
```
Comment (aymericdevriesere) [inline src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:391]: Les deux TODO dans les handlers sont pour une fonctionnalité à déterminer avec Sarah qui est en vacances !

Comment (qpic-cgi) [reply src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:391]: okey, du coup mais ta PR en DRAFT ou en WIP pour le moment

File: src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java [hunk 2]
```diff
@@ -295,15 +302,286 @@ public Future<JsonArray> uploadFiles(UserNextcloud.TokenProvider userSession, Li
...
+
+    /**
+     * @param userSession       User session
+     * @param user              User infos
+     * @param file              Path of the file you want to move
+     * @param result            Json where status of move is stored
+     * @param parentId          Id of the previous folder if moving in a folder
+     * @return                  Future Json with the infos about every move
+     */
+    private java.util.function.Function<JsonObject, Future<JsonObject>> moveHandler(UserNextcloud.TokenProvider userSession,
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:414]: ?
ça retourne pas directement Future<JsonObject> plutôt ?

File: src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java [hunk 3]
```diff
@@ -295,15 +302,286 @@ public Future<JsonArray> uploadFiles(UserNextcloud.TokenProvider userSession, Li
...
+                                                                             UserInfos user,
+                                                                             String file,
+                                                                             JsonObject result,
+                                                                             String parentId) {
+        return v -> {
```
Comment (qpic-cgi) [inline src/main/java/fr/openent/nextcloud/service/impl/DefaultDocumentsService.java:419]: créer une promesse et retourne là plutôt

**Comment by aymericdevriesere** [comment]
## Describe your changes
Adding two new routes to API. The first one will be used to move a file from Nextcloud to workspace and the second one will be used to copy a file from Nextcloud to workspace. Nextcloud and Workspace need to belong to the same user.
## Checklist tests
Tested with postman
## Issue ticket number and link
[ DRIV-26 ]
https://entsupport.gdapublic.fr/projects/DRIV/issues/DRIV-26
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequences regressions (must specify in **Checklist tests**)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequence feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

**Pull request closed without merging**